### PR TITLE
Feature/emis helpers

### PR DESCRIPTION
**Pull request by bbakernoaa**
helper functions for emission state and emission_mod

### CODE REVIEW

**Comment by bbakernoaa** [comment]
@lwcugb this includes the changes for the helper functions and such I was telling you about

**Comment by bbakernoaa** [comment]
@zmoon please take a Quick Look at this one

**Comment by bbakernoaa** [comment]
@zmoon bump

File: src/core/chemstate_mod.F90 [hunk 1]
```diff
@@ -1,11 +1,11 @@
-!> \file
+!> \file chemstate_mod.F90
```
Comment (zmoon) [inline src/core/chemstate_mod.F90]:
```suggestion
!>
```
Should default to the actual file name, no?

File: src/core/diagstate_mod.F90 [hunk 2]
```diff
@@ -17,14 +19,36 @@ module DiagState_Mod
...
+   !!
+   !! \ingroup core_modules
+   !!
+   !! \param dust_total_flux      Total flux of dust particles [kg m-2 s-1]
+   !! \param sea_salt_total_flux  Total flux of sea salt particles [kg m-2 s-1]
+   !! \param AOD550               Total AOD at 550nm [1]
+   !! \param AOD380               Total AOD at 380nm [1]
+   !! \param TOMSAI               TOMS Aerosol Index [1]
+   !! \param briggs_plumerise_height Effective plume rise height from Briggs algorithm [m]
+   !! \param sofiev_plumerise_height Effective plume rise height from Sofiev algorithm [m]
```
Comment (zmoon) [inline src/core/diagstate_mod.F90]: Is this needed? You're documenting them with `!<` below.

File: src/core/emisstate_mod.F90 [hunk 3]
```diff
@@ -74,21 +94,302 @@ module EmisState_Mod
    !! \ingroup core_modules
    !!
    !! \param State Name of the state
-   !! \param nEmisCategories Number of emission categories
-   !! \param EmisCats Emission category containers
+   !! \param nCats Number of emission categories
+   !! \param Cats Emission category containers
```
Comment (zmoon) [inline src/core/emisstate_mod.F90:98]: 😸 I guess it is CATChem, after all

File: src/core/emisstate_mod.F90 [hunk 4]
```diff
@@ -74,21 +94,302 @@ module EmisState_Mod
...
+         do t = 1, EmisState%nEmisTotal
+            ! Fill Total Species fluxes and properties
+            EmisState%TotSpecies(t)%name = EmisState%TotEmisNames(s)
+            EmisState%TotSpecies(t)%Flux = 0. ! set all fluxes to zero
+            currName = EmisState%TotEmisNames(s)
+            do c = 1, EmisState%nCats
+               do s = 1, EmisState%Cats(c)%nSpecies
+                  if (currName == EmisState%Cats(c)%Species(s)%name) then
+                     EmisState%TotSpecies(t)%Flux(:) = EmisState%TotSpecies(t)%Flux(:) +  &
+                        EmisState%Cats(c)%Species(s)%Flux(:)
```
Comment (zmoon) [inline src/core/emisstate_mod.F90:259]: I guess `EmisState%Cats(c)%Species(s)%Flux(:)` is nonzero? or at least could be?

File: src/core/emisstate_mod.F90 [hunk 5]
```diff
@@ -74,21 +94,302 @@ module EmisState_Mod
...
+      ! Error handling
+      CHARACTER(LEN=255) :: ErrMsg
+      CHARACTER(LEN=255) :: ThisLoc
+
+      ! Initialize
+      RC = CC_SUCCESS
+      ErrMsg = ''
+      ThisLoc = ' -> at Apply_Emis_to_Chem (in core/emisstate_mod.F90)'
+
+      ! TODO: implement this
```
Comment (zmoon) [inline src/core/emisstate_mod.F90:354]: Is this going to happen in this PR?

File: src/core/metstate_mod.F90 [hunk 6]
```diff
@@ -336,8 +277,8 @@ SUBROUTINE Met_Allocate( GridState, MetState, RC)
       MetState%PARDR = ZERO
       MetState%PARDF = ZERO
       MetState%PBLH = ZERO
-      MetState%PBL_TOP_hPa = ZERO
-      MetState%PBL_TOP_L = ZERO
+      ! MetState%PBL_TOP_hPa = ZERO
+      ! MetState%PBL_TOP_L = ZERO
```
Comment (zmoon) [inline src/core/metstate_mod.F90]: why commented?

**Review by zmoon — COMMENTED** [review]
Is this including some things from the plume rise branch?

Maybe some new tests can be added that verify this new functionality (cats and such).

File: src/core/diagstate_mod.F90 (hunk 2 above)
Comment (bbakernoaa) [reply src/core/diagstate_mod.F90]: I'm a little unclear if it is duplicated or not 🤷

File: src/core/emisstate_mod.F90 (hunk 4 above)
Comment (bbakernoaa) [reply src/core/emisstate_mod.F90:259]: it should always be zero or positive definite for the emission itself no?

File: src/core/emisstate_mod.F90 (hunk 5 above)
Comment (bbakernoaa) [reply src/core/emisstate_mod.F90:354]: no not in this PR.

File: src/core/chemstate_mod.F90 [hunk 7]
```diff
@@ -1,4 +1,4 @@
-!> \file chemstate_mod.F90
+!>
```
Comment (zmoon) [inline src/core/chemstate_mod.F90:1]: I think you still have to put `\file`(?), but not the file name itself, unless it's different from the file it's in

Comment (bbakernoaa) [reply src/core/chemstate_mod.F90:1]: I think I'm just going to add it back in.... we have it that way for all of the other files right now

**Review by zmoon — APPROVED** [review]

**Pull request merged**